File: frontend/logsense/src/app/model/Application.ts
import {EventData, Process, StatisticData} from "./PCData";
import {TimeMetrics} from "./TimeMetrics";

export class Application {
  pc: number = 0;
  application_name: string = "";
  time_series_data: ApplicationTimeSeriesData[] = [];
  cpu_events_and_anomalies: EventData[] = [];
  ram_events_and_anomalies: EventData[] = [];
  cpu_statistic_data: StatisticData = new StatisticData();
  ram_statistic_data: StatisticData = new StatisticData();
  run_time_in_seconds: AppTime = new AppTime();
  info: ApplicationInfo = new ApplicationInfo();
}

export class AppTime {
  name: string = "";
  total_running_time_seconds: number = 0;
}
export class ApplicationTimeSeriesData {
  measurement_time: string = "";
  cpu: number = 0;
  ram: number = 0;
}

export class ApplicationInfo {
  process_id: number = 0;
  path: string = "";
  working_directory: string = "";
  command_line: string = "";
  windows_user_name: string = "";
  bitness: number = 0;
  state: string = "";
  major_faults: number = 0;
  context_switches: number = 0;
  threads: number = 0;
  open_files: number = 0;
}

File: backend/FastApiRest/model/application.py
from __future__ import annotations
from pydantic import BaseModel, Json
from typing import List
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field

from model.data import Justification, StatisticData


class ApplicationTimeSeriesData(BaseModel):
    measurement_time: datetime
    cpu: float
    ram: float


class TimeMetricData(BaseModel):
    name: str
    total_running_time_seconds: float


class TimeMetricDataList(BaseModel):
    data: List[TimeMetricData]


class ApplicationInfo(BaseModel):
    process_id: int
    path: str
    working_directory: str
    command_line: str
    windows_user_name: str
    bitness: int
    state: str
    major_faults: int
    context_switches: int
    threads: int
    open_files: int


class ApplicationListObject(BaseModel):
    pc_id: int
    start: datetime
    end: datetime
    application_list: List[str]


class ApplicationData(BaseModel):
    pc: int
    application_name: str
    time_series_data: List[ApplicationTimeSeriesData]
    cpu_events_and_anomalies: List[Justification]
    ram_events_and_anomalies: List[Justification]
    cpu_statistic_data: StatisticData
    ram_statistic_data: StatisticData
    run_time_in_seconds: TimeMetricData
    info: ApplicationInfo